File: modules/file-transfer/src/main/java/com/gmail/marcosav2010/fth/packet/PacketFileSendFailed.java
package com.gmail.marcosav2010.fth.packet;

import com.gmail.marcosav2010.communicator.packet.Packet;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder;
import com.gmail.marcosav2010.fth.FileTransferHandler.FileSendResult;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.io.IOException;

@AllArgsConstructor
@NoArgsConstructor
public class PacketFileSendFailed extends Packet {

    @Getter
    private int fileID;
    @Getter
    private FileSendResult cause;

    @Override
    protected void encodeContent(PacketEncoder out) throws IOException {
        out.write(fileID);
        out.write((byte) cause.ordinal());
    }

    @Override
    protected void decodeContent(PacketDecoder in) throws IOException {
        fileID = in.readInt();
        cause = FileSendResult.values()[in.readByte()];
    }
}

File: modules/file-transfer/src/main/java/com/gmail/marcosav2010/fth/packet/PacketFileSendFailed.kt
package com.gmail.marcosav2010.fth.packet

import com.gmail.marcosav2010.communicator.packet.Packet
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder
import com.gmail.marcosav2010.fth.FileTransferHandler.FileSendResult
import kotlin.properties.Delegates

class PacketFileSendFailed() : Packet() {

    var fileID by Delegates.notNull<Int>()
        private set

    var cause by Delegates.notNull<FileSendResult>()
        private set

    constructor(fileID: Int, cause: FileSendResult) : this() {
        this.fileID = fileID
        this.cause = cause
    }

    override fun encodeContent(out: PacketEncoder) {
        out.write(fileID)
        out.write(cause.ordinal.toByte())
    }

    override fun decodeContent(`in`: PacketDecoder) {
        fileID = `in`.readInt()
        cause = FileSendResult.values()[`in`.readByte().toInt()]
    }
}